packages/monitoring-system/data-processor/test/unit/data-processors/mocks/octokit-middleware.ts

```diff
@@ -0,0 +1,173 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import {Octokit} from '@octokit/rest';
+import {resolve} from 'path';
+import {
+  OctokitRequestParser,
+  GitHubActionType,
+  GitHubObjectType,
+  GitHubActionDetails,
+  OctokitRequestOptions,
+} from './octokit-request-parser';
+
+/**
+ * Parsed details of GitHub action
+ */
+interface GitHubActionDetailsStrict {
+  type: GitHubActionType;
+  value: string;
+  destObjType: GitHubObjectType;
+  destObjId: string | number;
+  repoName: string;
+  repoOwner: string;
+}
+
+export interface MockResponse {
+  type: 'resolve' | 'reject';
+  value: {} | [];
+}
+
+interface MockResponses {
+  [type: string]: {
+    [value: string]: {
+      [dstObjType: string]: {
+        [destObjId: string]: {
+          [repoName: string]: {
+            [repoOwner: string]: MockResponse;
+          };
+        };
+      };
+    };
+  };
+}
+
+/**
+ * A middleware to intercept outgoing Octokit requests and
+ * return predefined responses instead.
+ */
```

Comment:
Can we add documentation/simple usage example?

Reply:
Sure! For now I'll add it within the file itself since we're not publishing/exporting it. When we do, I'll add usage guides in the README